bundles/org.openhab.binding.pixometer/src/main/java/org/openhab/binding/pixometer/handler/MeterHandler.java

```diff
@@ -0,0 +1,254 @@
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No handling needed, since we have only values to read.
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Pixometer handler '{}'", getThing().getUID());
+        updateStatus(ThingStatus.UNKNOWN);
+
+        try {
+            PixometerMeterConfiguration config = getConfigAs(PixometerMeterConfiguration.class);
+            setRessourceID(config.resource_id);
+
+            Bridge b = this.getBridge();
+            if (b == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                        "Could not find bridge (pixometer config). Did you choose one?");
+                return;
+            }
+
+            String token = new StringBuilder("Bearer ")
+                    .append(b.getConfiguration().get(CONFIG_BRIDGE_AUTH_TOKEN).toString()).toString();
+
+            obtainMeterId(token);
+
+            // Start polling job with the interval, that has been set up in the bridge
+            int pollingPeriod = Integer.parseInt(b.getConfiguration().get(CONFIG_BRIDGE_REFRESH).toString());
+            pollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                logger.debug("Try to refresh meter data");
+                try {
+                    updateMeter(token);
+                } catch (RuntimeException r) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR);
+                }
+            }, 2, pollingPeriod, TimeUnit.MINUTES);
+            logger.debug("Refresh job scheduled to run every {} hours. for '{}'", pollingPeriod, getThing().getUID());
+        } catch (RuntimeException r) {
+            logger.debug("Caught exception in ScheduledExecutorService of BridgeHandler. RuntimeException: ", r);
+            logger.debug("Could not initialize Thing {}: ", getThing().getUID(), r);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+        }
+        super.dispose();
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("Bridge Status updated to {} for device: {}", bridgeStatusInfo.getStatus(), getThing().getUID());
+        if (bridgeStatusInfo.getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, bridgeStatusInfo.getDescription());
+        }
+    }
+
+    /**
+     * Requests the corresponding meter data and stores the meterId internally for later usage
+     *
+     * @param token The current active auth token
+     */
+    private void obtainMeterId(String token) {
+        try {
+            String url = getApiString(API_METER_ENDPOINT);
+
+            Properties urlHeader = new Properties();
```

Comment:
For better design and keep the handler classes smaller put all api specific code in a separate class.

Reply:
Good point.
I would like to do that in one run when moving api related parts into the bridge.